amplify/backend/function/expressServer/src/controllers/leagueController.js

```diff
@@ -0,0 +1,90 @@
+const League = require('../models/league.model')
+
+// @desc create new league. a league is created and sent to the league-data
+// collection in the DB. the user's id is sent through the auth middleware,
+// this id is added to the users array as the first user in the league
+// @route post league/createLeague
+// @access private
+
+const createLeague = async (req, res, next) => {
+    try {
+        // parse the body and destructure
+        const {
+            leagueName,
+            budget,
+            leagueType,
+            tradingFee,
+            portfolios
+          } = req.body
+
+    // check they have sent all fields
+     if (
+      typeof leagueName === 'undefined' ||
+      typeof budget === 'undefined' ||
+      typeof leagueType === 'undefined' ||
+      typeof tradingFee === 'undefined' 
+    ) {
+      // data is missing bad request
+      res.status(400)
+      res.errormessage = 'All details are needed to create a league'
+      return next(
+        new Error(
+          'The client has not sent the required information to create a league',
+        ),
+      )
+    } 
+    let league = await League.findOne({ leagueName })
```

Comment:
See this too, I think the leagueName should not be unique, people will eventually call things the same name like two groups of different friends will probably have an "ESG League", the ID in database I think is enough to be unique.